catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java

```diff
@@ -43,44 +47,66 @@
 import ddf.catalog.util.impl.ServiceComparator;
 
 @Command(scope = CatalogCommands.NAMESPACE, name = "migrate", description = "Migrates Metacards "
-        + "from a Configured External Provider into the Catalog.  The migrate command currently "
-        + "picks the first Catalog provider as the FROM and the second Catalog provider as the TO. "
-        + "Additional support will be added to specify which provider will be migrated TO and FROM "
-        + "in the future.")
+        + "from a Configured External Provider to another Provider.")
 public class MigrateCommand extends DuplicateCommands {
     private static final Logger LOGGER = LoggerFactory.getLogger(MigrateCommand.class);
 
+    @Option(name = "--list", required = false, aliases = {
+            "-list"}, multiValued = false, description = "Print a list of external providers before migrating.")
+    boolean listProviders = false;
+
+    @Option(name = "--from", required = false, aliases = {
+            "-from"}, multiValued = false, description = "The Source ID of the external provider to migrate from.")
+    String fromProviderId;
+
+    @Option(name = "--to", required = false, aliases = {
+            "-to"}, multiValued = false, description = "The Source ID of the external provider to migrate to.")
+    String toProviderId;
+
     @Override
     protected Object executeWithSubject() throws Exception {
         if (batchSize > MAX_BATCH_SIZE || batchSize < 1) {
             console.println("Batch Size must be between 1 and 1000.");
             return null;
         }
 
-        List<CatalogProvider> providers = getCatalogProviders();
+        final List<CatalogProvider> providers = getCatalogProviders();
+
+        if (listProviders) {
+            if (providers.size() == 0) {
+                console.println("There are no available providers.");
+                return null;
+            }
+            console.println("Available providers:");
+            providers.stream()
+                    .map(p -> p.getClass()
+                            .getSimpleName())
+                    .forEach(id -> console.println("\t" + id));
```

Comment:
Should we return on this and end the operation when the `--list` option is used? I can't imagine something like `catalog:migrate --list -to x -from y` being done.